src/v/kafka/data/record_batcher.cc

```diff
@@ -17,7 +17,7 @@
 #include "storage/record_batch_builder.h"
 #include "utils/human.h"
 
-namespace kafka::client {
+namespace kafka::data {
```

Comment:
makes sense to move the batcher out of kafka/client. but is the batcher even specific to the kafka layer? it kind of seems like it's a form of a record_batch_builder (storage) or record batch utils (model/record_utils).

not that we need to find the perfect place for it in this pr, just curious about the scope of the batcher really.